Upload directory and multiple files, show upload progress bar, and keyboard-control upload menu

Levilian opened the pull request:
Implement #1050

Levilian committed:
folder and multiple file upload

percyliang commented:
Seems like we have a lot of files added unintentionally.  Probably best to recreate the branch to not have these in our history.

yipenghe commented:
Lol there still several unwanted files (#codalab_service.py# ...etc.

Levilian committed:
progress bar added and file size and length limit client-side check implemented

yipenghe commented:
There seems to be some very big changes, we should test it thoroughly on dev before merging to master

percyliang commented:
Are we sure all the original extraneous files are actually purged from the history?  I'd feel safer having a new branch...

Levilian commented:
> Are we sure all the original extraneous files are actually purged from the history? I'd feel safer having a new branch...

Yup! I rebased the branch and squashed the commits

yipenghe reviewed (commented):
Using click works great!

Keyboard shortcut  (:0)doesn't seem to work, "a u" will trigger the menu to show, but "up" "down" keyboards will move both the focus on worksheet and the upload menu. Hitting "enter" also doesn't open the upload menu, but will trigger bundle detail / text editing in the worksheet area

yipenghe commented:
I also like the percentage progress!

Levilian committed:
menu directly below the upload button

Levilian committed:
partial solution: keyboard shortcuts for dir/file upload

epicfaace reviewed (changes requested):
Great work so far!

Levilian committed:
Merge branch 'master' into upload-dir

yipenghe committed:
Merge branch 'master' into upload-dir

Levilian committed:
Merge branch 'master' into upload-dir

Levilian committed:
upload menu keyboard shortcuts

Levilian committed:
fix travis

Levilian committed:
Merge branch 'master' into upload-dir

Levilian committed:
removed empty lines

Levilian committed:
Merge branch 'master' into upload-dir

Levilian committed:
Merge branch 'master' into upload-dir
